crates/driver/src/infra/blockchain/token.rs

```diff
@@ -0,0 +1,85 @@
+use {super::Error, crate::domain::eth};
+
+/// An ERC-20 token.
+///
+/// https://eips.ethereum.org/EIPS/eip-20
+pub struct Erc20 {
+    contract: contracts::ERC20,
+}
+
+impl Erc20 {
+    pub(super) fn new(contract: contracts::ERC20) -> Self {
+        Self { contract }
+    }
+
+    /// Returns the [`eth::TokenAddress`] of the ERC20.
+    pub fn address(&self) -> eth::TokenAddress {
+        self.contract.address().into()
+    }
+
+    /// Fetch the ERC20 allowance for the spender. See the allowance method in
+    /// EIP-20.
+    ///
+    /// https://eips.ethereum.org/EIPS/eip-20#methods
```

Comment:
I guess the links above the function implementations could either be dropped or modified to link to the actual function on the website.

Reply:
They point to the `methods` section (Which includes all ERC-20 contract method documentation). It was mostly inherited from the `allowance` function documentation.

Reply:
Yes. But it's also possible to link to the exact method that is getting implemented.

Reply:
🤯

Reply:
Fixed.